client/src/components/Profile/EducationCard.tsx

```diff
@@ -1,162 +1,154 @@
-import * as React from 'react';
-import isEqual from 'lodash/isEqual';
+import React, { ChangeEvent, useState } from 'react';
 import { Typography, List, Input, Button } from 'antd';
+import { ReadOutlined, FileAddOutlined, DeleteOutlined } from '@ant-design/icons';
 import CommonCard from './CommonCard';
-import { GeneralInfo } from 'common/models/profile';
-import { ConfigurableProfilePermissions } from 'common/models/profile';
-import { ChangedPermissionsSettings } from 'pages/profile';
-import { CheckboxChangeEvent } from 'antd/lib/checkbox';
 
 const { Text } = Typography;
 
-import { ReadOutlined, FileAddOutlined, DeleteOutlined } from '@ant-design/icons';
+type UniversityProps = {
+  university: string | null;
+  faculty: string | null;
+  graduationYear: number | string | null;
+};
 
 type Props = {
-  data: GeneralInfo;
+  data: UniversityProps[];
   isEditingModeEnabled: boolean;
-  permissionsSettings?: ConfigurableProfilePermissions;
-  onPermissionsSettingsChange: (event: CheckboxChangeEvent, settings: ChangedPermissionsSettings) => void;
   onProfileSettingsChange: (event: any, path: string) => void;
+  saveProfile: () => void;
 };
 
-class EducationCard extends React.Component<Props> {
-  private filterPermissions = ({ isEducationVisible }: Partial<ConfigurableProfilePermissions>) => ({
-    isEducationVisible,
-  });
+const EducationCard = ({
+  isEditingModeEnabled,
+  data: educationHistory,
+  onProfileSettingsChange,
+  saveProfile,
+}: Props) => {
+  const [universities, setUniversities] = useState(educationHistory);
+  const isDisabled =
+    !!universities.length &&
+    !universities.every(({ university, faculty, graduationYear }) => university && faculty && graduationYear);
```

Comment:
I can suggest variant with a little bit less amount of operations
```suggestion
    !(universities.length ||
    universities.every(({ university, faculty, graduationYear }) => university && faculty && graduationYear));
```

Reply:
will not do - in your case if `universities.length` is truthy it won't check fields, but I need to check both - length and that some field is empty to set `isDisabled = true`. Come to think of it, I'll switch to `universities.some` it'd be slightly better.